app/models/post.rb

```diff
@@ -29,6 +30,10 @@ def text_post?
     text.present?
   end
 
+  def users_vote(user)
+    self.votes.find_by(user_id: user.id)
+  end
```

Comment:
could this be a `has_many` on `User`?

Reply:
I'm not sure if this is meant to be taken at face value or is a leading question. I have now set up a `has_many(votes)` association on User. Were you suggesting I could rewrite this method as a feature of the User class? I use this method to evaluate how the vote buttons should be displayed depending on whether the current user has cast a vote for the particular post that's being evaluated. Since I'm going to see how I can simplify the conditionals I have in the views, I may not need this method, but wanted to ask about it in case there might be a better way to get the info I needed.

Reply:
I agree, I think we can re-evaluate this comment when you've revisited the views. I have a hunch that this can actually live on the `Vote` model, since this is trying to look up a specific Vote, but won't know for sure until the dust settles on the views. Let me know if you want to pair on those at all.